app/plugin_api_test.go

```diff
@@ -375,35 +375,107 @@ func TestPluginAPISavePluginConfig(t *testing.T) {
+		expectedPluginConfigJsonString string
+		pluginConfig                   map[string]interface{}
+		actualConfig                   map[string]interface{}
+		expectedConfig                 map[string]interface{}
+	}
+
+	tests := []pluginTest{
+		{
+			name: "Plugin settings without default values provided in manifest",
+			manifest: &model.Manifest{
+				Id: "pluginid",
+				SettingsSchema: &model.PluginSettingsSchema{
+					Settings: []*model.PluginSetting{
+						{Key: "MyStringSetting", Type: "text"},
+						{Key: "MyBoolSetting", Type: "bool", Default: false},
+					},
+				},
+			},
+			pluginConfigJsonString:         `{"mystringsetting": "str", "myboolsetting": ""}`,
+			expectedPluginConfigJsonString: `{"mystringsetting": "str", "myboolsetting": false}`,
+		},
+		{
+			name: "Plugin settings with default value provided in manifest",
+			manifest: &model.Manifest{
+				Id: "pluginid",
+				SettingsSchema: &model.PluginSettingsSchema{
+					Settings: []*model.PluginSetting{
+						{Key: "MyStringSetting", Type: "text", Default: "defaultValue"},
+						{Key: "MyDefaultSetting", Type: "generated", Default: "defaultValue"},
+					},
+				},
+			},
+			pluginConfigJsonString:         `{"mystringsetting": "", "mydefaultsetting": ""}`,
+			expectedPluginConfigJsonString: `{"mystringsetting": "defaultValue", "mydefaultsetting": "defaultValue"}`,
+		},
+		{
+			name: "Plugin settings with empty json as a plugin config",
+			manifest: &model.Manifest{
+				Id: "pluginid",
+				SettingsSchema: &model.PluginSettingsSchema{
+					Settings: []*model.PluginSetting{
+						{Key: "MyStringSetting", Type: "text", Default: "defaultValue"},
+					},
+				},
+			},
+			pluginConfigJsonString:         `{}`,
+			expectedPluginConfigJsonString: `{"mystringsetting": "defaultValue"}`,
+		},
+		{
+			name: "Plugin settings with default value provided in manifest",
+			manifest: &model.Manifest{
+				Id: "pluginid",
+				SettingsSchema: &model.PluginSettingsSchema{
+					Settings: []*model.PluginSetting{
+						{Key: "MyStringSetting", Type: "text"},
+						{Key: "MyDropdownSetting", Type: "dropdown", Default: "defaultDropdownValue"},
+					},
+				},
 			},
+			pluginConfigJsonString:         `{"mystringsetting": "str", "mydropdownsetting": ""}`,
+			expectedPluginConfigJsonString: `{"mystringsetting": "str", "mydropdownsetting": "defaultDropdownValue"}`,
+		},
+		{
+			name: "Plugin settings with default value in manifest not override from provided configuration",
+			manifest: &model.Manifest{
+				Id: "pluginid",
+				SettingsSchema: &model.PluginSettingsSchema{
+					Settings: []*model.PluginSetting{
+						{Key: "MyStringSetting", Type: "text", Default: "defaultSettings"},
+					},
+				},
+			},
+			pluginConfigJsonString:         `{"mystringsetting": "str"}`,
+			expectedPluginConfigJsonString: `{"mystringsetting": "str"}`,
 		},
 	}
+	for _, testCase := range tests {
+		testCase.api = NewPluginAPI(th.App, testCase.manifest)
```

Comment:
Please use [sub tests](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks) to run this. That helps with debugging if a test fails.